crates/api-desc/src/scheduling.rs

```diff
@@ -22,20 +22,19 @@ pub(crate) fn new() -> Item {
             /// Waits until a callback is scheduled.
             ///
             /// This can be used as power management, since the CPU will sleep while waiting.
-            fn wait_for_callback "sw" {} -> {}
+            ///
+            /// Returns zero on success.
+            fn wait_for_callback "sw" {}
         },
         item! {
             /// Returns how many callbacks are pending.
-            fn num_pending_callbacks "sh" {} -> {
-                /// How many callbacks are pending.
-                count: usize,
-            }
+            fn num_pending_callbacks "sh" {}
         },
         item! {
             /// Aborts the applet.
             ///
-            /// This call doesn't return.
-            fn abort "sa" {} -> {}
+            /// Does not return on success.
+            fn abort "sa" {}
```

Comment:
It might be worth trying to statically ensure some properties on the resulting type:
- `-> !` only returns negative values
- `-> ()` does not return positive values
- `-> bool` only returns values less than 2
- `-> usize` returns all possible `isize`

Besides making `api-desc` slightly more readable, this could add some additional type-checking in the scheduler.